frontend/src/vault/keygen/shared/peerDiscovery/KeygenPeerDiscoveryStep/KeygenPeerDiscoveryStep.tsx

```diff
@@ -0,0 +1,102 @@
+  ComponentWithBackActionProps,
+  ComponentWithDisabledState,
+  ComponentWithForwardActionProps,
+  TitledComponentProps,
+} from '../../../../../lib/ui/props';
+import { QueryDependant } from '../../../../../lib/ui/query/components/QueryDependant';
+import { Query } from '../../../../../lib/ui/query/Query';
+import { StrictText } from '../../../../../lib/ui/text';
+import { PageContent } from '../../../../../ui/page/PageContent';
+import { PageHeader } from '../../../../../ui/page/PageHeader';
+import { PageHeaderBackButton } from '../../../../../ui/page/PageHeaderBackButton';
+import { PageHeaderTitle } from '../../../../../ui/page/PageHeaderTitle';
+import { useVaultType } from '../../../../setup/shared/state/vaultType';
+import { KeygenNetworkReminder } from '../../KeygenNetworkReminder';
+import { PeersManager } from '../../PeersManager';
+import { DownloadKeygenQrCode } from '../DownloadKeygenQrCode';
+import { KeygenPeerDiscoveryQrCode } from '../KeygenPeerDiscoveryQrCode';
+import { ManageServerType } from '../ManageServerType';
+import { Content, Wrapper } from './KegenPeerDiscoveryStep.styled';
+
+type KeygenPeerDiscoveryStepProps = ComponentWithForwardActionProps &
+  Partial<ComponentWithBackActionProps> &
+  TitledComponentProps &
+  ComponentWithDisabledState & {
+    joinUrlQuery: Query<string>;
+  };
+
+export const KeygenPeerDiscoveryStep = ({
+  onForward,
+  onBack,
+  title,
+  isDisabled,
+  joinUrlQuery,
+}: KeygenPeerDiscoveryStepProps) => {
+  const { t } = useTranslation();
+  const vaultType = useVaultType();
+
+  return (
+    <>
+      <PageHeader
+        title={<PageHeaderTitle>{title}</PageHeaderTitle>}
+        primaryControls={<PageHeaderBackButton onClick={onBack} />}
+        secondaryControls={
+          <QueryDependant
+            query={joinUrlQuery}
+            success={value => <DownloadKeygenQrCode value={value} />}
+            error={() => null}
+            pending={() => null}
+          />
+        }
+      />
+      <PageContent
+        as="form"
+        {...getFormProps({
+          onSubmit: onForward,
+          isDisabled,
+        })}
+      >
+        <Wrapper>
+          <Content>
+            <QueryDependant
+              query={joinUrlQuery}
+              success={value => <KeygenPeerDiscoveryQrCode value={value} />}
+              pending={() => (
+                <TakeWholeSpaceCenterContent>
+                  <Spinner />
+                </TakeWholeSpaceCenterContent>
+              )}
+              error={() => (
+                <TakeWholeSpaceCenterContent>
+                  <StrictText>{t('failed_to_generate_qr_code')}</StrictText>
+                </TakeWholeSpaceCenterContent>
+              )}
+            />
+
+            <VStack gap={40} alignItems="center">
+              {vaultType === 'secure' && <ManageServerType />}
+              <PeersManager />
+              <KeygenNetworkReminder />
+            </VStack>
+          </Content>
+          <DynamicInfoPanel
```

Comment:
<img width="1091" alt="image" src="https://github.com/user-attachments/assets/064e8c5d-4281-4417-9411-59023a737618">
Vultisig Logo , and "Join Keygen" , these should only show up in the export image , not on the screen when doing keygen